ChuckFacts/Fact/View/Error/Connection/ConnectionErrorViewModel.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  ConnectionErrorViewModel.swift
+//  ChuckFacts
+//
+//  Created by Calebe Emerick on 12/03/2018.
+//  Copyright © 2018 Stone Pagamentos. All rights reserved.
+//
+
+import UIKit
+
+final class ConnectionErrorViewModel {
```

Comment:
Nomes são sempre um problema, rs

Quando eu vejo `ConnectionErrorViewModel`, dependendo do contexto, eu já penso que essa classe reflete a camada de cima do ViewController, e não uma estrutura de dados dedicada a esse estado da interface.

Às vezes, nos meus projetos, eu opto por fazer como você fez aqui; mas o que seria equivalente ao `FactsViewModel` - e que contém a API pública que me permite recuperar os fatos - eu costumo chamar de `FactsScreen`. Dessa forma, eu sei que existe a correspondência 1 Screen <-> 1 ViewController, mas os estados em si na visão (ViewModels) podem ser eventualmente diversos, de acordo com a estratégia que eu for adotar para renderizar esses estados.

Não sei se me fiz entender aqui

Reply:
Não tinha percebido isso, mas dependendo do contexto parece mesmo ser uma camada de tratamento acima do `FactViewModel`. Acha que `ConnectionErrorScreen` ficaria ok? Já que uso `View`no final pra simbolizar que é do tipo `UIView`.

Reply:
Tendo uma diferença de propósitos, para mim tanto faz